src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -94,28 +94,26 @@ public void bootstrapApi() throws Exception {
             collection.setEntity(entity);
 
             api.addCollection(collection);
-        }
+        });
     }
 
-    public AmazonKinesisFirehoseAsync getFirehoseClient() {
-        if (this.firehoseClient == null) {
-            synchronized (this) {
```

Comment:
I'm not sure what the original programmer was thinking here - maybe that they didn't have a synchronized keyword available for their function?  Anyway, using the sugar.  Put synchronized on the function and don't check twice.